packages/ui-text-input/src/TextInput/index.tsx

```diff
@@ -140,6 +140,17 @@ class TextInput extends Component<TextInputProps, TextInputState> {
       })
     }
 
+    if (
+      getInteraction({ props: { interaction: prevProps.interaction } }) !==
+        'disabled' &&
+      getInteraction({ props: { interaction: this.props.interaction } }) ===
+        'disabled'
+    ) {
+      this.setState({
+        hasFocus: false
```

Comment:
I would also tell Omar that putting the focus back on the TextInput after the overlay is gone should happen on their side, and it can be easily achieved with the public `focus()` method on the component.

```tsx
  // Get the ref in the render method:
  render() {
    return (
      <div position="relative">
        <Select
          ref={(textInput)=> { this._input = textInput }}
          {...}
        </Select>
     </div>
    )
  }

  // use the focus method after the overlay is gone
  componentDidUpdate(prevState) {
    if (
      prevState.isOverlayActive !== this.state.isOverlayActive &&
      this.state.isOverlayActive === true
    ) {
      setTimeout(() => {
        this.setState({ isOverlayActive: false }, () => {
          this._input.focus()
          console.log("active element after", document.activeElement);
        });
      }, 2000);
    }
  }
```

Reply:
Yeah, I will mention this, but it seems, that he doesn't have an issue with the focus management, only with the bug.